### PR TITLE
Rename Progress to ProgressBar

### PR DESCRIPTION
**Pull request by tomi-msft**
This PR is to rename `Progress` to `ProgressBar` to be more inline with the broader component space

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25929/perf-test-react-components/Avatar-mount_base.html">1540</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25929/perf-test-react-components/Avatar-mount.html">1599</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25929/perf-test-react-components/Button-mount_base.html">1123</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25929/perf-test-react-components/Button-mount.html">1161</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25929/perf-test-react-components/FluentProvider-mount_base.html">1883</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25929/perf-test-react-components/FluentProvider-mount.html">1903</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25929/perf-test-react-components/FluentProviderWithTheme-mount_base.html">774</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25929/perf-test-react-components/FluentProviderWithTheme-mount.html">764</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25929/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">696</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25929/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">696</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25929/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">735</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25929/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">736</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25929/perf-test-react-components/MakeStyles-mount_base.html">2235</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25929/perf-test-react-components/MakeStyles-mount.html">2285</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25929/perf-test-react-components/Persona-mount_base.html">3667</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25929/perf-test-react-components/Persona-mount.html">3508</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25929/perf-test-react-components/SpinButton-mount_base.html">2957</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25929/perf-test-react-components/SpinButton-mount.html">3068</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the 26 screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-fluent/index.html?organization=uifabric&project=fabricpublic&prId=25929&commitId=dd92b452a519c72b7ea7f8b6f09bc4ca9f0c9a1c&env=prod)

✅ There was 13 screenshots added, 13 screenshots removed, 1746 screenshots unchanged, 0 screenshots with different dimensions and 0 screenshots with visible difference.


  <details><summary><b>unknown</b> 26 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|ProgressBar converged.Determinate - Dark Mode.chromium.png|0| Added|
|ProgressBar converged.Determinate - High Contrast.chromium.png|0| Added|
|ProgressBar converged.Determinate - RTL.chromium.png|0| Added|
|ProgressBar converged.Determinate with thickness large.chromium.png|0| Added|
|ProgressBar converged.Determinate.chromium.png|0| Added|
|ProgressBar converged.Error.chromium.png|0| Added|
|ProgressBar converged.Indeterminate - Dark Mode.chromium.png|0| Added|
|ProgressBar converged.Indeterminate - High Contrast.chromium.png|0| Added|
|ProgressBar converged.Indeterminate - RTL.chromium.png|0| Added|
|ProgressBar converged.Indeterminate with thickness large.chromium.png|0| Added|
|ProgressBar converged.Indeterminate.chromium.png|0| Added|
|ProgressBar converged.Success.chromium.png|0| Added|
|ProgressBar converged.Warning.chromium.png|0| Added|
|Progress converged.Determinate - Dark Mode.chromium.png|0| Removed|
|Progress converged.Determinate - High Contrast.chromium.png|0| Removed|
|Progress converged.Determinate - RTL.chromium.png|0| Removed|
|Progress converged.Determinate with thickness large.chromium.png|0| Removed|
|Progress converged.Determinate.chromium.png|0| Removed|
|Progress converged.Error.chromium.png|0| Removed|
|Progress converged.Indeterminate - Dark Mode.chromium.png|0| Removed|
|Progress converged.Indeterminate - High Contrast.chromium.png|0| Removed|
|Progress converged.Indeterminate - RTL.chromium.png|0| Removed|
|Progress converged.Indeterminate with thickness large.chromium.png|0| Removed|
|Progress converged.Indeterminate.chromium.png|0| Removed|
|Progress converged.Success.chromium.png|0| Removed|
|Progress converged.Warning.chromium.png|0| Removed|

</details>


<div id="vrtCommentfluentuiv9"/>

File: packages/react-components/react-progress/src/components/ProgressField/ProgressField.tsx [hunk 1]
```diff
@@ -7,15 +7,15 @@ import {
   useField_unstable,
 } from '@fluentui/react-field';
 import type { ForwardRefComponent } from '@fluentui/react-utilities';
-import { Progress } from '../../Progress';
+import { ProgressBar } from '../../ProgressBar';
 
-export type ProgressFieldProps = FieldProps<typeof Progress>;
+export type ProgressFieldProps = FieldProps<typeof ProgressBar>;
```
Comment (spmonahan) [inline packages/react-components/react-progress/src/components/ProgressField/ProgressField.tsx:12]: Should this be `ProgressBarFieldProps`? And the file name changed as well?

File: packages/react-components/react-progress/stories/ProgressBar/ProgressBarBestPractices.md [hunk 2]
```diff
@@ -2,11 +2,11 @@
 
 ### Do
 
-- Use an `indeterminate` Progress when the total units to completion is unknown
+- Use an `indeterminate` ProgressBar when the total units to completion is unknown
```
Comment (spmonahan) [inline packages/react-components/react-progress/stories/ProgressBar/ProgressBarBestPractices.md:5]:
```suggestion
- Use an `indeterminate` `ProgressBar` when the total units to completion is unknown
```

File: packages/react-components/react-components/etc/react-components.unstable.api.md [hunk 3]
```diff
@@ -135,14 +135,14 @@ import { personaClassNames } from '@fluentui/react-persona';
 import { PersonaProps } from '@fluentui/react-persona';
 import { PersonaSlots } from '@fluentui/react-persona';
 import { PersonaState } from '@fluentui/react-persona';
-import { Progress } from '@fluentui/react-progress';
-import { progressClassNames } from '@fluentui/react-progress';
+import { ProgressBar } from '@fluentui/react-progress';
```
Comment (spmonahan) [inline packages/react-components/react-components/etc/react-components.unstable.api.md:138]: I believe we should maintain the deprecated `Progress` exports here until `ProgressBar` moves to stable or did the plan change (or maybe I misunderstood)?

File: packages/react-components/react-progress/src/components/ProgressField/ProgressField.tsx (hunk 1 above)
Comment (tomi-msft) [reply packages/react-components/react-progress/src/components/ProgressField/ProgressField.tsx:12]: We are keeping it as `ProgressField`

File: packages/react-components/react-components/etc/react-components.unstable.api.md (hunk 3 above)
Comment (tomi-msft) [reply packages/react-components/react-components/etc/react-components.unstable.api.md:138]: No I accidentally removed them, I will add them back, good callout

File: packages/react-components/react-components/src/unstable/index.ts [hunk 4]
```diff
@@ -363,10 +363,20 @@ export {
...
+  // eslint-disable-next-line deprecation/deprecation
   useProgress_unstable,
+  ProgressBar,
+  progressBarClassNames,
+  renderProgressBar_unstable,
+  useProgressBarStyles_unstable,
+  useProgressBar_unstable,
 } from '@fluentui/react-progress';
-export type { ProgressProps, ProgressState, ProgressSlots } from '@fluentui/react-progress';
+export type { ProgressBarProps, ProgressBarState, ProgressBarSlots } from '@fluentui/react-progress';
```
[the remark this answers is not in the document]
Comment (tomi-msft) [reply packages/react-components/react-components/src/unstable/index.ts]: Ahh good callout, I added them, thank you

**Review by behowell — APPROVED** [review]
@sopranopillow 
> It would be great to have an issue filed so that when releasing ProgressBar, we remove the deprecated portions of Progress.

Good idea! @tomi-msft could you log this issue and add TODO comments by each of the deprecated exports that references that issue?

Like this, but put in the actual issue number:
```
// TODO #99999: Remove deprecated export before ProgressBar is released as stable
/** @deprecated renamed to ProgressBar */
export const Progress = ProgressBar;
```

**Pull request merged**